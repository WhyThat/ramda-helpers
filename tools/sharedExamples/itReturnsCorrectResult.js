export default (result) => {
  test('returns correct result', () => {
    expect(result).toMatchSnapshot();
  });
};
