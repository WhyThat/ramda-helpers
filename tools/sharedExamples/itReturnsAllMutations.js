import itReturnsCorrectResult from './itReturnsCorrectResult';

export default (fn, mutations) => {
  mutations.forEach((mutation, index) => {
    const name = mutation.name || `mutations ${index}`;

    describe(name, () => {
      itReturnsCorrectResult(fn(mutation.props));
    });
  });
};
