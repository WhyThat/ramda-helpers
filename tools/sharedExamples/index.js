
import itReturnsAllMutations from './itReturnsAllMutations';
import itReturnsCorrectResult from './itReturnsCorrectResult';

export {
  itReturnsAllMutations,
  itReturnsCorrectResult,
};
