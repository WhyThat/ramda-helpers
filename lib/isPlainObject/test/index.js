import { itReturnsAllMutations } from '../../test/sharedExamples';
import { isPlainObject, isNotPlainObject } from '../';

describe('isPlainObject', () => {
  const mutations = [{
    name: 'With null',
    props: null,
  }, {
    name: 'With array',
    props: [],
  }, {
    name: 'With object',
    props: {},
  }, {
    name: 'With function',
    props: () => {},
  }];

  itReturnsAllMutations(isPlainObject, mutations);
});

describe('isNotPlainObject', () => {
  const mutations = [{
    name: 'With null',
    props: null,
  }, {
    name: 'With array',
    props: [],
  }, {
    name: 'With object',
    props: {},
  }, {
    name: 'With function',
    props: () => {},
  }];

  itReturnsAllMutations(isNotPlainObject, mutations);
});

