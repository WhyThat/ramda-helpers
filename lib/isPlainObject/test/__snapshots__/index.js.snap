// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isNotPlainObject With array returns correct result 1`] = `true`;

exports[`isNotPlainObject With function returns correct result 1`] = `true`;

exports[`isNotPlainObject With null returns correct result 1`] = `true`;

exports[`isNotPlainObject With object returns correct result 1`] = `false`;

exports[`isPlainObject With array returns correct result 1`] = `false`;

exports[`isPlainObject With function returns correct result 1`] = `false`;

exports[`isPlainObject With null returns correct result 1`] = `false`;

exports[`isPlainObject With object returns correct result 1`] = `true`;
